src/commands/emulators-exec.ts

```diff
@@ -12,12 +12,17 @@ import { DatabaseEmulator } from "../emulator/databaseEmulator";
 import { EmulatorRegistry } from "../emulator/registry";
 import { FirestoreEmulator } from "../emulator/firestoreEmulator";
 import * as commandUtils from "../emulator/commandUtils";
+import * as getProjectId from "../getProjectId";
 
-async function runScript(script: string): Promise<number> {
+async function runScript(script: string, projectId: string | undefined): Promise<number> {
   utils.logBullet(`Running script: ${clc.bold(script)}`);
 
   const env: NodeJS.ProcessEnv = { ...process.env };
 
+  if (projectId) {
```

Comment:
What if they already set `GCLOUD_PROJECT` on the outside?  We should probably not override.

Reply:
I think the one by `emulators:exec` is usually what people want since if I'd expect `firebase emulators:exec './seed-db.sh'` to automatically write to the right project and database, say if I'm using the Python Admin SDK. If they really want to overwrite, they can always do `firebase emulators:exec 'GCLOUD_PROJECT=foo' ./script`. So I think it's the same story with the emulator host variables in that sense.